From a68a4b446338e249cc24ac2b15069ea13521ee89 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Thu, 19 Apr 2018 14:31:07 +0200
Subject: [PATCH 0595/1051] arm64: dts: marvell: 8040-db: add comphy references
 to SATA ports

Adding phandles to the SATA ports will allow AHCI driver to configure
appropriate marvell, comphy-cp110 accordingly.

Change-Id: I8a1043cea8560840b04bfe7fbc237b337e91c218
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../arm64/boot/dts/marvell/armada-8040-db.dts | 26 +++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index 9d40aa676c33..d3700da9906b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -158,9 +158,20 @@
 
 };
 
-/* CON4 on CP0 expansion */
 &cp0_sata0 {
 	status = "okay";
+	/* CON2 on CP0 expansion */
+	sata-port@0 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy1 0>;
+	};
+	/* CON4 on CP0 expansion */
+	sata-port@1 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy3 1>;
+	};
 };
 
 /* CON9 on CP0 expansion */
@@ -295,9 +306,20 @@
 	};
 };
 
-/* CON4 on CP1 expansion */
 &cp1_sata0 {
 	status = "okay";
+	/* CON2 on CP1 expansion */
+	sata-port@0 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp1_comphy1 0>;
+	};
+	/* CON4 on CP1 expansion */
+	sata-port@1 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp1_comphy3 1>;
+	};
 };
 
 /* CON9 on CP1 expansion */
-- 
2.17.1

